src/services/accession.jl

```diff
@@ -0,0 +1,461 @@
+        write(io, '.')
+        write(io, string(getversion(ccds.data)))
+    end
+end
+
+@doc """Return a URI for a Consensus CDS entry.""" ->
+function uri(ccds::Accession{:CCDS}; format::Symbol=:browser)
+    @assert format === :browser
+    return URI("http://www.ncbi.nlm.nih.gov/CCDS/CcdsBrowse.cgi?REQUEST=CCDS&DATA=$ccds")
+end
+
+
+# Ensembl
+# -------
+
+# webpage:
+#   http://www.ensembl.org/index.html
+# accession format:
+#   http://www.ensembl.org/info/genome/stable_ids/index.html
+
+function ismatch(::Type{Accession{:Ensembl}}, s::String)
+    return ismatch(r"^\s*(:?ENS(:?[A-Z]{3,4}?)?|FB)(:?E|FM|G|GT|P|R|T)\d+(:?\.\d+)?\s*$", s)
+    #                    ^~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~^~~^~~~~~~~~
+    #                    species prefix            feature prefix     |  version (optional)
+    #                                                                 numerals
+end
+
+@doc """
+Parse an accession number string of Ensembl (e.g. "ENSG00000141510").
+""" ->
+function parse(::Type{Accession{:Ensembl}}, s::String)
+    @check_match :Ensembl s
+    dot = search(s, '.')
+    if dot == 0
+        accession = strip(s)
+        version = Nullable{Uint8}()
+    else
+        accession = strip(s[1:dot-1])
+        version = Nullable(parse(Uint8, s[dot+1:end]))
+    end
+    return Accession{:Ensembl,Versioned{ASCIIString}}(Versioned{ASCIIString}(accession, version))
+end
+
+function show(io::IO, ensembl::Accession{:Ensembl})
+    write(io, ensembl.data.accession)
+    if isversioned(ensembl.data)
+        write(io, '.')
+        write(io, string(getversion(ensembl.data)))
+    end
+    return
+end
+
+# TODO: Is there a permanent link to an entry?
+#function uri(ensembl::Accession{:Ensembl}; format::Symbol=:browser)
+#end
+
+
+# Gene Ontology
+# -------------
+
+# webpage:
+#   http://geneontology.org/
+# accession format:
+#   http://geneontology.org/page/ontology-structure#termstru
+
+function ismatch(::Type{Accession{:GeneOntology}}, s::String)
+    return ismatch(r"^\s*GO:\d{7}\s*$", s)
+end
+
+@doc """
+Parse an accession number string of Gene Ontology (e.g. "GO:0030330").
+""" ->
+function parse(::Type{Accession{:GeneOntology}}, s::String)
+    @check_match :GeneOntology s
+    go = search(s, "GO:")
+    n = parse(Uint32, s[last(go)+1:end])
+    return Accession{:GeneOntology,Uint32}(n)
+end
+
+function show(io::IO, go::Accession{:GeneOntology})
+    @printf io "GO:%07d" convert(Uint32, go.data)
+end
+
+@doc """
+Return a URI for a Gene Ontology entry.
+""" ->
+function uri(go::Accession{:GeneOntology}; format::Symbol=:browser)
+    if format === :browser
+        q = "id=$go"
+    else
+        @assert format ∈ [:mini, :obo, :oboxml]
+        q = "id=$go&format=$format"
+    end
+    # TODO: AmiGO or QuickGO
+    return URI("http://www.ebi.ac.uk/QuickGO/GTerm?$q")
+end
+
+
+# UniProt
+# -------
+
+# webpage:
+#   http://www.uniprot.org/
+# accession format:
+#   http://www.uniprot.org/help/accession_numbers
+
```

Comment:
At the moment, No. I'd like to have a one-to-one mapping between identifiers and entries that is stable and primary accession number.
But the entry name of UniProt would be useful to access the database, so it should be supported as a database-specific package/module in the future.
Thank you for pointing :+1: